openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -1999,21 +2000,27 @@ struct AMDGPUDeviceTy : public GenericDeviceTy, AMDGenericDeviceTy {
 
     // TODO: We should try to avoid materialization but there seems to be no
     // good linker interface w/o file i/o.
-    SmallString<128> LinkerOutputFilePath;
-    std::error_code EC = sys::fs::createTemporaryFile(
-        "amdgpu-pre-link-jit", ".out", LinkerOutputFilePath);
+    SmallString<128> LinkerInputFilePath;
+    std::error_code EC = sys::fs::createTemporaryFile("amdgpu-pre-link-jit",
```

Comment:
Does this "guarantee" uniqueness in a multi-threaded / multi-process environment?

Reply:
No but that could only happen when multiple libraries are initialized all at once, or users explicitly dlopen libraries with OpenMP target offloading.

Reply:
I don't think it's an issue, the underlying code more or less just tries random filenames until one works. There may be an astronomically slim chance that the stars align and both threads create the same random name at the same exact time and none of the other code detects it, but my gut says that the Linux kernel would handle that fine.

Reply:
That's sound as much of a guarantee as I was hoping for.